src/GafferUI/AnimationGadget.cpp

```diff
@@ -267,6 +461,17 @@ AnimationGadget::AnimationGadget()
 
 AnimationGadget::~AnimationGadget()
 {
+	m_selectedKeys->clear();
```

Comment:
What's the reason for clearing explicitly here (and the associated assert in `~SelectionSet`)? I'm wondering if it'll bite us due to the emission of `memberRemovedSignal()` - depending on the order of destruction of the components of AnimationEditor, could it lead to one component handling the signal and then trying to access the now-dead AnimationGadget?

Reply:
> the selection set can outlive the animation gadget, so it makes sense to clear the selected keys

I'm not really sure what the benefit of clearing is though - if something is keeping a hold of the set after the AnimationGadget has died, wouldn't it be more useful to keep the contents? Since the set is shared (held by Ptr), perhaps the AnimationGadget doesn't have sole rights to clear it?

> My primary concern was if an exception was thrown from the signal handlers, but that gets caught anyway I believe by the catchingsignalcombiner stuff.

Looks like we're not actually using CatchingSignalCombiner in `Set::MemberSignal` - some of this older stuff predates `CatchingSignalCombiner` and `translatePythonException()` and hasn't been updated. Could you make a separate PR to bring that up to date please? It should be a case of adding the combiner and updating `MemberSignalSlotCaller` in SetBinding.cpp to use `translatePythonException()` rather than `PyErr_PrintEx`.

Reply:
>  Could you make a separate PR to bring that up to date please?

issue created https://github.com/GafferHQ/gaffer/issues/4402